tests/unit/test_configuration.py

```diff
@@ -185,27 +183,105 @@ def test_format_miswrite_ages(user_config_type, tmp_path):
             "column_noise": {
                 "age": {
                     "age_miswriting": {
-                        "possible_perturbations": [-2, -1, 2],
+                        "possible_perturbations": user_config,
                     },
                 },
             },
         },
     }
-    if user_config_type == "path":
-        filepath = tmp_path / "user_dict.yaml"
-        with open(filepath, "w") as file:
-            yaml.dump(user_config, file)
-        user_config = filepath
-
-    new_dict = get_configuration(
-        user_config
-    ).decennial_census.column_noise.age.age_miswriting.to_dict()
-    default_dict = (
-        get_configuration().decennial_census.column_noise.age.age_miswriting.to_dict()
-    )
-    assert default_dict["row_noise_level"] == new_dict["row_noise_level"]
-    # check that 1 got replaced with 0 probability
-    assert new_dict["possible_perturbations"][1] == 0
-    # check that others have 1/3 probability
-    for p in [-2, -1, 2]:
-        assert new_dict["possible_perturbations"][p] == 1 / 3
+
+    config = get_configuration(user_config)[FORMS.census.name][Keys.COLUMN_NOISE][
+        COLUMNS.age.name
+    ][NOISE_TYPES.age_miswriting.name][Keys.AGE_MISWRITING_PERTURBATIONS].to_dict()
+
+    assert config == expected
+
+
+@pytest.mark.parametrize(
+    "config, match",
+    [
+        ({"fake_form": {}}, "Invalid form '.*' provided. Valid forms are "),
+        (
+            {FORMS.acs.name: {"other_noise": {}}},
+            "Invalid configuration key '.*' provided for form '.*'. ",
+        ),
+        (
+            {FORMS.acs.name: {Keys.ROW_NOISE: {"fake_noise": {}}}},
+            "Invalid noise type '.*' provided for form '.*'. ",
+        ),
+        (
+            {FORMS.acs.name: {Keys.ROW_NOISE: {NOISE_TYPES.omission.name: {"fake": {}}}}},
+            "Invalid parameter '.*' provided for form '.*' and noise type '.*'. ",
+        ),
+        (
+            {FORMS.acs.name: {Keys.COLUMN_NOISE: {"fake_column": {}}}},
+            "Invalid column '.*' provided for form '.*'. ",
+        ),
+        (
+            {FORMS.acs.name: {Keys.COLUMN_NOISE: {COLUMNS.age.name: {"fake_noise": {}}}}},
+            "Invalid noise type '.*' provided for form '.*' for column '.*'. ",
+        ),
+        (
+            {
+                FORMS.acs.name: {
+                    Keys.COLUMN_NOISE: {
+                        COLUMNS.age.name: {NOISE_TYPES.missing_data.name: {"fake": 1}}
+                    }
+                }
+            },
+            "Invalid parameter '.*' provided for form '.*' for column '.*' and noise type '.*'",
+        ),
+    ],
+    ids=[
```

Comment:
I don't understand what this is

Reply:
The `ids` kwarg gives a name to the parameterized tests. 
![image](https://user-images.githubusercontent.com/11376379/232106113-cde0f409-670a-4b4a-b0e5-f96356886ecf.png)